examples/common/datt-config/DattConfig.cpp

```diff
@@ -0,0 +1,29 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <DattConfig.h>
+
+__attribute__((weak)) const chip::Credentials::AttestationTrustStore * GetAttestationTrustStore()
```

Comment:
Why is this needed? Can't an application just pass a different object to `SetDeviceAttestationCredentialsProvider` instead of `chip::Credentials::Examples::GetExampleDACProvider()`?

Reply:
Currently, if a developer wants to override the default DAC Provider, he will need to edit the example code. What if the example code is updated? The developer will need to maintain his example code up-to-date constantly.

This PR is an attempt to fix that and to make example tools modular, so that the developer can later simply link an example lib (lighting-app lib in this case) against his implementation of the two weak methods introduced in this PR.